### ISSUE TITLE
SD card download not possible 

### ISSUE THREAD

**Issue by TheHenry455** [futo-org/grayjay-android#861]
Cannot download to the SD card even though the folder in SD card has been selected via the available option in settings. 
It seems to be downloading straight to local storage , which is very limited.

**Comment by wedsa5** [comment]
Yes, it still seems to be an issue. I am not sure where the file actually stored. It must be in the application data directory or something.
The setting suggests to the user that all downloaded content for offline viewing will be placed in the selected folder, not stored within app data/internal storage.
Only when the downloaded video is exported is it put into the selected external download folder.

Screenshots:

As you can see here, I set the external downloads folder (and general) to the SD card:
![image](https://github.com/user-attachments/assets/9de898ca-2d23-4b90-9baa-b8f5ffe8f782)
Then, I download a video:
![image](https://github.com/user-attachments/assets/e0336bf1-b786-4238-9e7f-6ac67c28015a)
And you can see the available space is roughly what my file manager displays for the internal storage, not the external storage:
![image](https://github.com/user-attachments/assets/bf5ecb7a-210c-4393-8a3a-52da6e70ff6a)
And once the file is completely downloaded, the external downloads folder is empty:
![image](https://github.com/user-attachments/assets/40da122b-d3f9-4b7e-a5f1-bcda54c678fd)
BUT, if I click to export the video, it will then be exported to the external downloads folder:
![image](https://github.com/user-attachments/assets/9cd4c4f0-ddad-472f-a6f3-2807907e9166)
![image](https://github.com/user-attachments/assets/12522470-a764-4ebe-b83e-f4a9de0134b6)

**Comment by Zvonimir-FUTO** [comment]
@wedsa5 this is normal behavior only when you export a video file, it will be saved as mp4.

**Comment by wedsa5** [comment]
> @wedsa5 this is normal behavior only when you export a video file, it will be saved as mp4.

The issue is that the downloaded videos (ignore exported) are not stored where the user chooses. Everything is stored on the internal memory, even when an SD card directory is selected.
As the OP points out, on phones with limited internal storage, this can become an issue, especially if you download many videos/a playlist as it will eat up all your internal storage space. 

The expected behavior is that if you choose to "Change external Downloads directory" to an SD card directory, that that's where the downloads will be, saving your internal storage space.
Also, the issue with Clearing external Downloads directory button not doing anything was not addressed and could be considered a bug on its own, but I believe it's related to the main bug.

If what you say is true and that is expected behavior, then the wording on the buttons in the settings needs to be changed to something like "Change **exported** downloads directory".

**Comment by RainOfPain125** [comment]
I'm having this issue as well. Google Pixel 8 Pro doesn't come with much storage, so I opted to buy a flash drive (because it doesn't come with an SD card slot either...).

However, there is no option in the app to choose whoch directory or directories that I want Grayjay to read for files, or to save files to.

There is the option (mentioned by people in this post) for designating "external Downloads directory" but this does not serve that purpose. This appears to only be for "exporting" ie COPYING already-downloaded media from internal storage, to the external storage.

Indeed, if you remove your external storage, the video is still downloaded (in internal storage) and can still play. If you delete the video from internal storage, and reconnect your external storage, the video is no longer in your downloads to be played within the Grayjay app. It is simply a plain .mp4 video sitting in external storage, and the Grayjay app does not "point to" or know about the video.

//

What I believe the issue is, is that this isn't a "bug" but a missing feature that seems like it should obviously be added. Users should be able to designate

1. what directory downloads will be saved to by default.
2. what directories Grayjay should search for downloaded content (in case there is more than the default internal storage, or designated external storage).
3. option for per-video or per-playlist download location designation.

@wedsa5 worded it perfectly.

> [@Zvonimir-FUTO](https://github.com/Zvonimir-FUTO)
> 
> > [@wedsa5](https://github.com/wedsa5) this is normal behavior only when you export a video file, it will be saved as mp4.
> 
> The expected behavior is that if you choose to "Change external Downloads directory" to an SD card directory, that that's where the downloads will be, saving your internal storage space.
> 
> If what you say is true and that is expected behavior, then the wording on the buttons in the settings needs to be changed to something like "Change **exported** downloads directory".

**Comment by nbhirud** [comment]
Probably a duplicate of https://github.com/futo-org/grayjay-android/issues/286

**Comment by domifi** [comment]
> Probably a duplicate of https://github.com/futo-org/grayjay-android/issues/286

No, no duplicate. The solution to [#286](https://github.com/futo-org/grayjay-android/issues/286) allows exporting *already downloaded* videos to an SD card. What this issue wants, is the option to store downloads directly on the sd card: You press download in Grayjay > the video gets downloaded to the SD card > you can watch the downloaded video in grayjay.
This isn't necessarily to allow third apps to access the files, but to save space on the internal storage.

**Comment by Zvonimir-FUTO** [comment]
Closing in favor of https://github.com/futo-org/grayjay-android/issues/3262
